core/state_transition.go

```diff
@@ -308,15 +308,6 @@ func (st *StateTransition) TransitionDb(refunds bool, gasBailout bool) (*Executi
 	}
 	st.gas -= gas
 
-	// Check clause 6
-	var bailout bool
-	if !msg.Value().IsZero() && !st.evm.Context.CanTransfer(st.state, msg.From(), msg.Value()) {
```

Comment:
I think this can only be removed if all transactions were "Dynamic Fee" transactions. For legacy transactions, this check IMO is still required. Do you think it will simply blow up later?

Reply:
> I think this can only be removed if all transactions were "Dynamic Fee" transactions. For legacy transactions, this check IMO is still required. Do you think it will simply blow up later?

This check is not required even for legacy transactions because a stronger version of it is already performed (after PR #2415) by `buyGas` (invoked by `preCheck`).